Switch LGPL'd chardet for MIT licensed charset_normalizer (#5797)
psf/requests: HISTORY.md

-   \[Short description of non-trivial change.\]

**Dependencies**

- Switch chardet for the MIT-licensed charset_normalizer for Python3 to remove license
  ambiguity for projects bundling requests. Python2 still depends upon the
  LGPL'd module.

> 

Reviewer: Maybe precise that if chardet is installed, it will still be used as it could confuse users.

Reviewer 2: Added.

Reviewer: Thanks!

-   \[Short description of non-trivial change.\]

**Dependencies**

- Instead of `chardet`, use the MIT-licensed `charset_normalizer` for Python3
  to remove license ambiguity for projects bundling requests. If `chardet`
  is already installed on your machine it will be used instead of `charset_normalizer`
  to keep backwards compatibility.

  You can also install `chardet` while installing requests by
  specifying `[use_chardet_on_py3]` extra as follows:

>   ```shell
>   pip install "requests[use_chardet_on_py3]"

Reviewer 3: Something went wrong with that suggestion .. the ending triple quotes disappeared somehow, try this

```suggestion
    ```shell
    pip install "requests[use_chardet_on_py3]"
    ```
```

-   \[Short description of non-trivial change.\]

**Dependencies**

> - Instead of `chardet` use the MIT-licensed `charset_normalizer` for Python3
>   to remove license ambiguity for projects bundling requests. If `chardet`
>   is installed it is still going to be used instead of `charset_normalizer`
>   to keep backwards-compatibility.
> 
>   You can also install chardet while installing requests by
>   specifying `[use_chardet_on_py3]` extra
> 
> ```bash
>     pip install "requests[use_chardet_on_py3]"
> ```

Reviewer 3: ```suggestion
- Instead of `chardet`, use the MIT-licensed `charset_normalizer` for Python3
  to remove license ambiguity for projects bundling requests. If `chardet`
  is already installed on your machine it will be used instead of `charset_normalizer`
  to keep backwards compatibility.

  You can also install `chardet` while installing requests by
  specifying `[use_chardet_on_py3]` extra as follows:

  ```shell
  pip install "requests[use_chardet_on_py3]"
  ```
```
